zulu/datetime.py

```diff
@@ -567,6 +568,114 @@ def span(self, frame, count=1):
+    def span_range(cls, frame, start=None, end=None):
+        """Class method that helps in getting a range of time spans.
+
+        Args:
+            frame (str): A time frame. Ex: year, month, day, minute, etc.
+            start (datetime): A start datetime value.
+            end (datetime): An end datetime value.
+
+        Returns:
+            list: List of all time spans
+        """
+        # Validate the start datetime, end datetime and the given frame.
+        cls.validate_times(start, end)
+        cls.validate_frame(frame)
+
+        span_list = []
+        is_less_than_end_time = True
+
+        # This loop helps in bringing all the time spans with the given time
+        # frame.
+        while is_less_than_end_time:
+            # We use the span() method to bring the actual tuple with the
+            # given time frame.
+            time_span_tuple = cls.fromdatetime(start).span(frame)
+
+            # We change the value of start date to the floor value of the
+            # returned tuple from span() method.
+            start = cls.fromdatetime(time_span_tuple[1])
+
+            # If the changed start value is less than the given end datetime,
+            # we make the loop stop otherwise, we add up the returned tuple to
+            # the list
+            if start > end:
+                is_less_than_end_time = False
+            else:
+                start = start.shift(microseconds=1)
+                span_list.append(time_span_tuple)
+
+        return span_list
+
+    @classmethod
+    def range(cls, frame, start=None, end=None):
+        """Class method that helps in getting a range of time values from the
+        given start datetime and end datetime with the given frame value.
+
+        Args:
+            frame (str): A time frame. Ex: year, month, day, minute, etc.
+            start (datetime): A start datetime value.
+            end (datetime): An end datetime value.
+
+        Returns:
+            list: A list of datetime values ranging from the given start and
+                end datetimes.
+        """
+        cls.validate_times(start, end)
+        cls.validate_frame(frame)
+
+        if frame == 'century':
+            # When the given frame is 'century', we make the value base-lined
+            # to year value. So, the value will be 100.
+            frame_value = 100
+        elif frame == 'decade':
+            # When the given frame is 'decade', we make the value base-lined
+            # to year value. So, the value will be 10.
+            frame_value = 10
+        else:
+            # All the other frame values will be 1.
+            frame_value = 1
+
+        if frame in ['century', 'decade']:
+            frame = 'year'
+
+        # Get the plural frame name for making it use in shift() method.
+        plural_frame = '{0}s'.format(frame)
+        range_list = []
+        is_less_than_end_time = True
+
+        while is_less_than_end_time:
+            # pylint: disable=E1123
```

Comment:
Ok, probably want to  put the disable comment right before the part that needs it and add a comment after it explaining why the disable is there. I might look into it more to see if there's another way to "fix" it.

One thing I did notice is that you could store `step = {plural_frame: frame_value}` in a variable so you're not creating a new `dict` each time since it never changes.